File: examples/blink_ble.py
"""
Blink an LED over BLE.

Requires the BLE firmware build on the board and the ble extra installed:
    pip install thingbot-telemetrix[ble]
"""

import time

from thingbot_telemetrix import Telemetrix

# Scan for any board advertising as ThingBot-<mac>.
# To skip scanning, pass ble_address='AA:BB:CC:DD:EE:FF' instead.
board = Telemetrix(ble_name="ThingBot")

board.gpio().set_pin_mode_output(7)

try:
    while True:
        board.gpio().digital_write(7, 1)
        time.sleep(0.5)
        board.gpio().digital_write(7, 0)
        time.sleep(0.5)
except KeyboardInterrupt:
    board.shutdown()

File: thingbot-telemetrix-rust/examples/blink_ble.rs
//! Blink an LED over BLE.
//!
//! Requires the BLE firmware build on the board:
//!
//! ```bash
//! cargo run --example blink_ble --features ble
//! ```

use std::thread::sleep;
use std::time::Duration;

use thingbot_telemetrix_rust::Telemetrix;

fn main() -> thingbot_telemetrix_rust::Result<()> {
    println!("Scanning for a ThingBot BLE device...");
    let mut board = Telemetrix::connect_ble()?;

    board.gpio().set_output(7)?;

    for _ in 0..10 {
        board.gpio().digital_write(7, true)?;
        sleep(Duration::from_millis(500));
        board.gpio().digital_write(7, false)?;
        sleep(Duration::from_millis(500));
    }

    Ok(())
}
